app/page.tsx

```diff
@@ -1,29 +1,40 @@
 "use client";
 import { useQuery, useMutation } from "convex/react";
 import { api } from "../convex/_generated/api";
-import type { ChangeEvent } from "react";
-import type { Id } from "@/convex/_generated/dataModel";
+import TextEditor from "./components/TextEditor";
+
+import type { ReactQuillProps } from "react-quill";
+
+type OnChange = ReactQuillProps["onChange"];
 
 export default function Home() {
   const inputValue = useQuery(api.input.getInputValue);
 
   const mutateValue = useMutation(api.input.updateInputValue);
 
-  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const payload = {
-      id: event.target.attributes.getNamedItem("hidden-id")?.value as Id<"input">,
-      text: event.target.value
+  const handleOnChange: OnChange = (_value, _delta, _source, editor) => {
+    if (inputValue === undefined || inputValue === null) {
```

Comment:
We will need to address this at some point (but not in this PR) 😅 

Probably in #22

Reply:
I just added [#28](https://github.com/web-dev-cody-hackathon/100editors/issues/28) for this issue